sklearn/utils/estimator_checks.py

```diff
@@ -355,10 +357,38 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def gram_matrix_if_pairwise(X, estimator, kernel=linear_kernel):
+
+    if len(X.shape) == 1:
+        X = X.reshape(-1, 1)
+
+    if is_pairwise_metric(estimator):
+        # pairwise_distance() fails for certain versions of scipy
+        n_obs = X.shape[0]
+        X_std = (X - X.mean(axis=0)) / X.std(axis=0)
+        X_out = np.zeros(shape=(n_obs, n_obs))
+        for i in range(n_obs):
+            for j in range(n_obs):
+                dist = np.sum((X_std[i] - X_std[j]) ** 2) ** .5
+                X_out[i, j] = dist
+        return X_out
+
+    if is_pairwise(estimator):
+        return kernel(X, X)
+
+    return X
+
+
 def check_estimator_sparse_data(name, estimator_orig):
+
+    # Sparse precomputed kernels aren't supported
```

Comment:
is there a reason we should exclude them altogether? better would be to check that the estimator either (a) behaves correctly with sparse precomputed kernels; or (b) raises an appropriate error message in this case.

Reply:
This is not addressed yet, right? (but you gave +1?)